Please add protobuf chages to get it build on Apple Silicon M1

**Describe the bug**
There is an issue of protobuf that it does not get the correct artifact name, which cause the plugin does not build to Android on Apple silicon M1

**To Reproduce**
1. Add package to the project.
2. Build & Run on Android.
3. You would see 

`FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':reactive_ble_mobile:generateDebugProto'.
> Could not resolve all files for configuration ':reactive_ble_mobile:protobufToolsLocator_protoc'.
   > Could not find protoc-3.13.0-osx-aarch_64.exe (com.google.protobuf:protoc:3.13.0).
     Searched in the following locations:
         https://repo.maven.apache.org/maven2/com/google/protobuf/protoc/3.13.0/protoc-3.13.0-osx-aarch_64.exe

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 6s
Exception: Gradle task assembleDebug failed with exit code 1`

**Expected behavior**
it builds

**Smartphone / tablet**
 - Device:  Pixel2
 - OS: Android 10
 - Package version: 3.3.1+1, 4.0.0, 4.0.1

**Peripheral device**


**Additional context**
Please made the following changes to the gradle file.
`
protobuf {
    protoc {
        // a fix for m1
        if (osdetector.os == "osx") {
            artifact = 'com.google.protobuf:protoc:3.13.0:osx-x86_64'
        } else {
            artifact = 'com.google.protobuf:protoc:3.13.0'
        }
    }

    generateProtoTasks {
        all().each { task ->
            task.builtins {
                java {
                    option "lite"
                }
            }
        }
    }
}`

--- a/packages/reactive_ble_mobile/android/build.gradle
+++ b/packages/reactive_ble_mobile/android/build.gradle
@@ -13,8 +13,8 @@ buildscript {
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:7.0.2'
-        classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.14'
+        classpath 'com.android.tools.build:gradle:7.0.3'
+        classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.17'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
         classpath "io.gitlab.arturbosch.detekt:detekt-gradle-plugin:$detekt_version"
         classpath "de.mannodermaus.gradle.plugins:android-junit5:1.7.1.1"
@@ -79,7 +79,7 @@ detekt {
 
 protobuf {
     protoc {
-        artifact = 'com.google.protobuf:protoc:3.13.0'
+        artifact = 'com.google.protobuf:protoc:3.18.1'
     }
 
     generateProtoTasks {
@@ -96,7 +96,7 @@ protobuf {
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
     implementation 'com.polidea.rxandroidble2:rxandroidble:1.13.0'
-    implementation 'com.google.protobuf:protobuf-javalite:3.14.0'
+    implementation 'com.google.protobuf:protobuf-javalite:3.18.1'
     implementation 'io.reactivex.rxjava2:rxkotlin:2.4.0'
     implementation 'io.reactivex.rxjava2:rxandroid:2.1.1'
 

